modules/behavioural_qc/php/conflicts.class.inc

```diff
@@ -0,0 +1,151 @@
+<?php declare(strict_types=1);
+
+namespace LORIS\behavioural_qc;
+
+use LORIS\behavioural_qc\Provisioners\ConflictsProvisioner;
+use LORIS\Data\Filters\HasAnyPermissionOrUserSiteMatch;
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+use \LORIS\Middleware\ETagCalculator;
+
+/**
+ * The Conflicts Class.
+ *
+ * GET request method for endpoint in behavioural_qc module.
+ * Used for populating the Data Conflicts table.
+ *
+ * @category Behavioural
+ * @package  Loris
+ * @author   Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class Conflicts extends \NDB_Page implements ETagCalculator
+{
+    /**
+     * A cache of the endpoint results, so that it doesn't need to be
+     * recalculated for the ETag and handler.
+     */
+    private $_cache;
+
+    /**
+     * This function will return a json response.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        // Ensure GET request.
+        switch ($request->getMethod()) {
+        case 'GET':
+            return $this->_handleGET($request);
+        default:
+            return new \LORIS\Http\Response\JSON\MethodNotAllowed(
+                $this->allowedMethods()
+            );
+        }
+    }
+    /**
+     * Initialize setup, the extra values for the
+     * "Data Conflicts" form. (psc & errors)
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handleGET(ServerRequestInterface $request) : ResponseInterface
+    {
+        $filter      = new HasAnyPermissionOrUserSiteMatch(
+            array('access_all_profiles')
+        );
+        $provisioner = (new ConflictsProvisioner())->filter($filter);
+
+        $user = $request->getAttribute('user');
+
+        $data         = (new \LORIS\Data\Table())
+            ->withDataFrom($provisioner)
+            ->toArray($user);
+        $body         = array(
+            'data'         => $data,
+            'fieldOptions' => $this->_getFieldOptions($request),
+            'subprojects'  => \Utility::getSubprojectList(),
+        );
+        $this->_cache = new \LORIS\Http\Response\JsonResponse($body);
```

Comment:
This cache is never used

Reply:
@xlecours does the cache not get used for the ETag? I don't see it being used but just want to make sure because I don't know how the ETag really works.